### PR TITLE
Add allowedIPs in the API and gateway

### LINKED ISSUE #287: Remove misleading documentation about an `-a, --allowip=` command line option
There is documentation accessible via the command line (-h) and in the README.md of this repo that indicates an `allowip` command line option is available to configure which addresses a server will allow API connections from.

Apparently, this is not a valid option (perhaps the documentation is a hold-over from v1) and the correct way to do this is to set the Addresses.Gateway setting in the config file.

Expected results:
- mentions of the `allowip` flag are removed from the documentation
- it would be nice if there was some documentation that the `Addresses.Gateway` setting should be used to accomplish this (perhaps in general some documentation is needed on how to configure a node to work with a remote client)

### PR DESCRIPTION
**Pull request by cpacia**
closes #287

### CODE REVIEW

**Comment by drwasho** [comment]
LGTM

File: repo/testdata/config [hunk 1]
```diff
@@ -49,11 +49,15 @@
     "PrivKey": "testKey"
   },
   "Ipns": {
+    "QuerySize": 0,
```
Comment (tyler-smith) [inline repo/testdata/config]: @cpacia what is the reason for this change?

Comment (cpacia) [inline repo/testdata/config]: That test data needs to be a properly formatted object for the test to pass. So it needs to have the QuerySize in there.
